src/app/cli/src/coda.ml

```diff
@@ -30,10 +30,14 @@ let maybe_sleep _ = Deferred.unit
 
 [%%endif]
 
-let chain_id ~genesis_state_hash =
+let chain_id ~genesis_state_hash ~genesis_constants =
```

Comment:
It would introduce a dependency cycle. `Genesis_constants` doesn't depend on any coda libraries except for `blake2` and `module_version`. This is made on purpose so that it can be used anywhere and is the reason for defining the checked version of `Genesis_constants.Protocol.t` in `coda_base/protocol_constants_checked.ml`.